.github/workflows/ci.yaml

```diff
@@ -1,95 +1,148 @@
-
-      - name: Set up Python ${{ matrix.python-version }}
+      - name: Checkout code
+        uses: actions/checkout@v2
+      - name: Set up Python
         uses: actions/setup-python@v2
         with:
-          python-version: ${{ matrix.python-version }}
+          python-version: 3.9
+      - name: Install Poetry
+        uses: snok/install-poetry@v1.1.2
 
-      - name: Install dependencies
+      - name: Create venv cache
+        id: cache-venv
+        uses: actions/cache@v2.1.4
+        with:
+          path: venv
+          key: ${{ runner.os }}-venv-${{ hashFiles('poetry.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-venv-
+
+      - name: Create venv and install dependencies
+        if: steps.cache-venv.outputs.cache-hit != 'true'
         run: |
-          python -m pip install --upgrade pip
-          pip install flake8
-          if [[ -f requirements.txt ]]; then pip install -r requirements.txt; fi
+          python -m venv venv
+          . venv/bin/activate
+          poetry install
 
-      - name: Lint with flake8
+      - name: Black
         run: |
-          # stop the build if there are Python syntax errors or undefined names
-          flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics
-          # exit-zero treats all errors as warnings. Set line length to 80 chars max.
-          flake8 . --count --exit-zero --max-complexity=10 --max-line-length=80 --statistics
+          . venv/bin/activate
+          pre-commit run --hook-stage manual black --all-files
 
-  pytest:
+  isort:
     runs-on: ubuntu-latest
-    strategy:
-      matrix:
-        python-version: [3.7, 3.8, 3.9]
-      fail-fast: false
-
     steps:
-      - uses: actions/checkout@v2
-
-      - name: Set up Python ${{ matrix.python-version }}
+      - name: Checkout code
+        uses: actions/checkout@v2
+      - name: Set up Python
         uses: actions/setup-python@v2
         with:
-          python-version: ${{ matrix.python-version }}
+          python-version: 3.9
+      - name: Install Poetry
+        uses: snok/install-poetry@v1.1.2
 
-      - name: Install dependencies
+      - name: Create venv cache
+        id: cache-venv
+        uses: actions/cache@v2.1.4
+        with:
+          path: venv
+          key: ${{ runner.os }}-venv-${{ hashFiles('poetry.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-venv-
+
+      - name: Create venv and install dependencies
+        if: steps.cache-venv.outputs.cache-hit != 'true'
         run: |
-          python -m pip install --upgrade pip
-          pip install flake8
-          if [[ -f requirements_test.txt ]]; then pip install -r requirements_test.txt; fi
+          python -m venv venv
+          . venv/bin/activate
+          poetry install
 
-      - name: Run Pytest
+      - name: isort
         run: |
-          pytest
+          . venv/bin/activate
+          pre-commit run --hook-stage manual isort --all-files
```

Comment:
Why use pre-commit here? What is the advantage of it?

I'd just do `poetry run isort .` here.